### ISSUE TITLE
Plugin crashes Obsidian when opening some long files

### ISSUE THREAD

**Issue by rk-kontur** [bwydoogh/obsidian-force-view-mode-of-note#39]
I have imported quite a few bookx from EPUB to MD as I need them for reference and linking to chapters / passages. So far, everything has worked fine.

I like using your plugin to "switch" finished files of myself or such books to preview only but recently, after switching, Obsidian crashes.

This is a complete crash, I cannot use any buttons on the interface anymore. The only way to deal with it is:

1. Close Obsidian from WIndows 11 (rightclick on symbol in launcher)
2. Delete workspace.json in vault folder
3. Restart.

When I open the file with this plugin disable, it display fine. Once I re_enable and restart, Obsidian crashes again.

Oddly, the behaviour does not happen with every long file. I tried a couple and the crashes mostly happen.

At first, I thought that files imported from EPUB were only affected but this is also not generally the case - some open fine while others do not.

Can you look into that? If I can be of any help, please let me know.
I really like this plugin - it makes life so much easier.

THANK YOU!

**Comment by bwydoogh** [comment]
Hi,

I can look into this, but I would need an example. Could you share one with me?

**Comment by rk-kontur** [comment]
Thank you for your swift reply. Yes. I can send you the file. Can you send me an email address via PM?

**Comment by bwydoogh** [comment]
Please send to argent-06.basso@icloud.com

**Comment by rk-kontur** [comment]
thank you. Just sent the file.

Let me know if you find anything

**Comment by rk-kontur** [comment]
Oh and btw: I just tried the other plugin that is aiming at switching default reading mode. And it crashes in the very same way.

This one is what I mean:
https://github.com/AlexDavies8/obsidian-frontmatter-viewmode

EDIT: Accidentally Inserted link to your plugin. Above one is what I meant.

**Comment by bwydoogh** [comment]
So, I guess I may close this issue; could you leave feedback in the Obsidian forums (report a bug)? It looks like it is more of an Obsidian bug than something else. Or the file is too large for Obsidian to handle in preview mode.

**Comment by rk-kontur** [comment]
I am not sure this is an Obsidian bug. That the same behaviour is happening in the other plugin only suggests that the two approaches are doing the same thing.

As for the file being too large: Larger files are opening fine and actually, the file I sent you was created with Obsidian.

**Comment by bwydoogh** [comment]
May I know the specifications of your machine? As I replied to your email, I cannot reproduce this issue (on a MacBook Pro M1 Pro 32bg ram). Perhaps it is a combination of obsidian performance and the specifications of your machine?

**Comment by rk-kontur** [comment]
So. I have identified the real problem and it is NOT this plugin. It looked like it was your plugin because it caused the files to switch the display mode. That this happened with the other plugin, too, seemed to suggest a bug in Obsidian.

In fact, the crashing entirely disappeared when I disabled another plugin: Iconize is the culprit - it seems to crash Obsidian when the editor mode is switched. Once this plugin is disabled, your plugin just works fine.

I apologize for the confusion.
